timeout_tools/cli.py

```diff
@@ -274,5 +282,67 @@ def load_python_version(ws=None):
         return False
 
 
+def check_python_version_installed(python_version):
+    ###
+    # Check whether a version of python is already installed via pyenv
+    ###
+    py_ver_present = False
+    (status, result) = run('pyenv versions --bare --skip-aliases')
+    if status:
+        raise PyEnvFailure({"message": "Failed to run"})
+    for py_ver in result.replace(' ', '').split('\n'):
+        if py_ver == python_version:
+            py_ver_present = True
+            break
+
+    if not py_ver_present:
+        try:
+            check_python_version_available(python_version)
+            try:
+                install_python_version(python_version)
+            except PyEnvFailure as e:
+                print(e.args[0]['message'])
+                sys.exit(1)
+        except PyEnvFailure as e:
+            print(e.args[0]['message'])
+            sys.exit(1)
+    else:
+        print(f'- Python `{python_version}` is installed ✅')
+
+
+def check_python_version_available(python_version):
+    ###
+    # Check whether a version of python is already is available for install by pyenv
+    # if not exit and tell user to update pyenv installation
+    ###
+    py_ver_available = False
+    (status, result) = run('pyenv install --list')
+    if status:
+        raise PyEnvFailure({"message": "Failed to run"})
+    for py_ver in result.replace(' ', '').split('\n'):
+        if py_ver == python_version:
+            py_ver_available = True
+            break
+    if not py_ver_available:
+        print('''
+            Please update pyenv with latest versions of python by running:
+            cd ~/.pyenv/plugins/python-build/../.. && git pull && cd -
+        ''')
+        raise PyEnvFailure({"message": "Pyenv requires update"})
+    else:
+        print(f'- Python `{python_version}` is available for installation ✅')
+
+
+def install_python_version(python_version):
+    ###
+    # Use pyenv to install new version of python
+    ###
+    (status, _) = run(f'pyenv install {python_version}')
+    if status:
+        raise BaseException(f'Failed to install python version {python_version}')
+    else:
+        print(f'- Python `{python_version}` installation successful ✅')
```

Comment:
the first part of the message should print before the installs start and the tick be added when in succeeds